### PR TITLE
[style] Styling the Login (and Sign Up) pages

### PR DESCRIPTION
**Pull request by plaanty**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Styled the Login and Sign Up pages according to the [figma](https://www.figma.com/design/fPzdPfT77bx4wXHBkAQbj6/Adopt-an-Inmate?node-id=4407-530&m=dev).

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."
Here are some images of what it looks like on my screen for the Login and Signup pages.
NOTE: The fonts should work now! Let me know if they don't.

[image placeholder]: 
image

image

image

image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Sam:
Start the server on your local computer, then navigate to the Login and Sign Up pages by clicking on the associated links. Then, check to see that you are able to type in the boxes and it responds to it. Also check that you are able to click on the buttons (they don't have any interactivity yet, so they shouldn't send you to anything, but they should respond to your click).
Play around with different page sizes, I don't think it should respond to this too well (as a lot of it is hardcoded to be reflective of the current state of the Figma design), but I will work on this if need be.

Jinkang:
Should also test form validations on login and sign up. Test empty fields, invalid inputs, short password, unchecked ToS, already existing users, etc.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Sam:
- Once Austin puts it into autolayout, potentially will be easier to accommodate different screen sizes

Jinkang:
- Should have a separate screen to inform the user to confirm email

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
https://tailwindcss.com/docs/

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ethan-tam33

### CODE REVIEW

File: components/TextBoxField.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export function TextBox({ input, placeholder }: TextBoxProps) {
+  const [info, setInfo] = useState('');
+
+  return (
+    <div>
+      <input
+        type={input}
+        placeholder={placeholder}
+        value={info}
+        onChange={e => setInfo(e.target.value)}
```
[the remark this answers is not in the document]
Comment (plaanty) [reply components/TextBoxField.tsx]: I put the styling in a cva element, is this what you meant by just expose the HTML attribute in a component prop?
As far as I understand it, I'm using the `useState` to update the value seen in the textbox itself, with `value` and `onChange`. Is this a misunderstanding on my part?

File: components/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsChecked(event.target.checked);
+  };
+
+  return (
+    <div>
+      <input type="checkbox" checked={isChecked} onChange={handleChange} />
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (plaanty) [reply components/Checkbox.tsx]: I wanted to add custom styling to make it appear the color that is present in Austin's figma design. I didn't add any to this because at the time I wasn't sure how to do so.
Given that this sprint is just about the Login styling (technically), I'll prioritize those comments and then come back to the ones about the sign up page later, especially since there's likely to be issues with merging with Sai's sprint that set up some of the signup page stuff.

File: components/TextBoxField.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jinkang-0) [reply components/TextBoxField.tsx]: Can you clarify what you meant by a `cva` element? Do you mean the component/element uses a class name determined by a function returned by `cva`?

In any case, that is not what I meant. Exposing the HTML attributes means to allow expose the interface of HTML attributes (e.g. `type`, `placeholder`, `value`, `onChange`, etc) as a prop for this component (which serves as a wrapper anyway). That is to say, the user of this component can call something like `<TextBox value={...} onChange={...} />` and it would be valid. The example provided is not exhaustive. An exhaustive list of props can be passed using the `forwardRef` method (ref Button component) or by using the helper type `React.HTMLAttributes<...>`.

Here is an example for an input checkbox:
```ts
export default function Checkbox({ ...props }: Omit<React.HTMLAttributes<HTMLInputElement>, "type">) {
  return (
    <input type="checkbox" {...props} />
  )
}
```

This would take any and all props passed to the `Checkbox` component and pass them along to the `input` element. We use `Omit<..., ...>` helper type to disallow passing the `type` attribute, fixing it to a checkbox input.
